lib/uplink/object.go

```diff
@@ -69,12 +72,20 @@ type ObjectMeta struct {
 type Object struct {
 	// Meta holds the metainfo associated with the Object.
 	Meta ObjectMeta
+
+	metainfo *kvmetainfo.DB
+	streams  streams.Store
 }
 
 // DownloadRange returns an Object's data. A length of -1 will mean
 // (Object.Size - offset).
 func (o *Object) DownloadRange(ctx context.Context, offset, length int64) (io.ReadCloser, error) {
-	panic("TODO")
+	readOnlyStream, err := o.metainfo.GetObjectStream(ctx, o.Meta.Bucket, o.Meta.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	return stream.NewDownload(ctx, readOnlyStream, o.streams), nil
```

Comment:
Does this actually work? Isn't `o.streams` just nil here? Does something set that somewhere I don't see?

Reply:
Ahh, nevermind, I see, it's filled in by the bucket when opening the object.